extensions/async-extensions/build.gradle

```diff
@@ -0,0 +1,78 @@
+plugins {
+  alias(deps.plugins.publish)
+  alias(deps.plugins.dokka)
+  id("app.cash.sqldelight.multiplatform")
+}
+
+archivesBaseName = 'sqldelight-async-extensions'
+
+kotlin {
+  js()
+
+  sourceSets {
+    commonMain {
+      dependencies {
+        implementation project(':runtime:runtime')
+        implementation project(':runtime:runtime-async')
+        api deps.kotlin.coroutines.core
+      }
+    }
+    commonTest {
+      dependencies {
+        implementation deps.kotlin.test.common
+        implementation deps.kotlin.test.commonAnnotations
+        implementation deps.turbine
+      }
+    }
+    jvmTest {
+      dependencies {
+        implementation deps.kotlin.test.junit
+        implementation deps.kotlin.coroutines.test
+        implementation project(':drivers:sqlite-driver')
+      }
+    }
+    jsTest {
+      dependencies {
+        implementation deps.kotlin.test.js
+        implementation project(':drivers:sqljs-driver')
+      }
+    }
+    nativeMain {
+      dependencies {
+        implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:${deps.versions.kotlinCoroutines.get()}-native-mt") {
+          force = true
+        }
+      }
+    }
+    nativeTest {
+      dependencies {
+        implementation project(':drivers:native-driver')
+        implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:${deps.versions.kotlinCoroutines.get()}-native-mt") {
+          force = true
+        }
+      }
+    }
+  }
+
+  configure([targets.iosX64, targets.iosArm32, targets.iosArm64, targets.tvosX64, targets.tvosArm64, targets.watchosX86, targets.watchosX64, targets.watchosArm32, targets.watchosArm64, targets.macosX64, targets.linuxX64, targets.macosArm64, targets.iosSimulatorArm64, targets.watchosSimulatorArm64, targets.tvosSimulatorArm64]) {
+    compilations.test {
+      kotlinOptions.freeCompilerArgs += ['-linker-options', '-lsqlite3 -L/usr/lib/x86_64-linux-gnu -L/usr/lib -L/usr/lib64']
+    }
+  }
+
+  // mingwX86 target is not supported by kotlinx.coroutines, see https://github.com/Kotlin/kotlinx.coroutines/issues/2157
+  configure([targets.mingwX64]) {
+    compilations.test {
+      kotlinOptions.freeCompilerArgs += ["-linker-options", "-Lc:\\msys64\\mingw64\\lib -L$rootDir\\libs -lsqlite3".toString()]
+    }
+  }
+
+  sourceSets.matching { it.name.endsWith("Test") }.all {
+    it.languageSettings {
+      optIn('kotlin.time.ExperimentalTime')
+      optIn('kotlinx.coroutines.ExperimentalCoroutinesApi')
+    }
+  }
```

Comment:
```suggestion
```

no test sources so can remove

Reply:
I figured we'd probably want to add some kind of tests before merging which is why I left those parts in. I've just been focused on coming up with a solution to handling transactions in the wrapper first.